PixelHunter1995/Inputs/InputConfigParser.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework.Input;
+
+namespace PixelHunter1995.Inputs
+{
+    using KeyDisjunction = Dictionary<Either<Keys, MouseKeys>, SignalState>;
+    using KeyConjunction = HashSet<Dictionary<Either<Keys, MouseKeys>, SignalState>>;
+    
+    class InputConfigParser
+    {
+        
+        public static readonly string DEFAULT_CONTEXT = "DEFAULT";
+        
+        public static Dictionary<string, Input> ParseInputConfig(string path)
+        {
+            var inputs = new Dictionary<string, Input>();
+            foreach (var kv in ParseKeyConfig(path)) {
+                var context = kv.Key;
+                var binds = kv.Value;
+                inputs[context] = new Input(binds);
+            }
+            return inputs;
+        }
+        
+        private static Dictionary<string, Dictionary<Action, KeyConjunction>> ParseKeyConfig(string path) {
+            
+            var contexts = new Dictionary<string, Dictionary<Action, KeyConjunction>>();
+            
+            if (File.Exists(path))
+            {
+                string[] lines = File.ReadAllLines(path);
+                foreach (string line in lines)
+                {
+                    string context;
+                    string actionStr;
+                    
+                    string[] kv = line.Split('=');
+                    string lhs = kv[0].Trim();
+                    
+                    // Make it possible to load keybinds for specific "contexts"
+                    // Could potentially be used for ie. secondary players,
+                    // or states.
```

Comment:
What happens if you put the same key with a different context? Will both actions be triggered?

Reply:
Depends on whether the Input instances for each context has both of their Update method called in the same frame. They do not prevent each other in any way, but are also not interlinked in any way (so updating one does not update the other). If used for states, two states could both have an action for escape, but only the active state is updating its input instance (or has its Update hook run at all to begin with).

This context functionality is currently not used, as general consensus (when asked in voip) was to _not_ have one input instance per state. I removed that functionality, but left the context-code just in case we changed our mind, but it could at this point be removed entirely if desired.